app/src/main/java/nmd/nomad/PlacesListActivity.java

```diff
@@ -0,0 +1,111 @@
+package nmd.nomad;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
+
+import java.util.ArrayList;
+
+import nmd.nomad.adapters.PlacesListArrayAdapter;
+import nmd.nomad.api.GomadClient;
+import nmd.nomad.api.ServiceGenerator;
+import nmd.nomad.models.Place;
+import retrofit.Callback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+public class PlacesListActivity extends ActionBarActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_places_list);
+
+        // Set the listView and the adapter
+        ProgressBar progressBar = (ProgressBar) findViewById(R.id.progressBar);
+        ListView listView = (ListView) findViewById(R.id.placesListView);
+        LinearLayout noPlaceFound = (LinearLayout) findViewById(R.id.noPlaceFound);
+        final ArrayList<Place> placeList = new ArrayList<Place>();
+        final PlacesListArrayAdapter arrayAdapter = new PlacesListArrayAdapter(this, placeList);
+        listView.setAdapter(arrayAdapter);
+
+        // Set a callback to fill the listView
+        final Callback callback = new Callback() {
+            @Override
+            public void success(Object object, Response response) {
+                ArrayList<Place> places = (ArrayList<Place>) object;
+                progressBar.setVisibility(View.GONE);
+
+                if(places.isEmpty()) {
+                    listView.setVisibility(View.GONE);
+                    noPlaceFound.setVisibility(View.VISIBLE);
+                } else {
+                    noPlaceFound.setVisibility(View.GONE);
+                    listView.setVisibility(View.VISIBLE);
+                    arrayAdapter.clear();
+                    arrayAdapter.addAll(places);
+                    for (Place place : places) {
+                        System.out.println(place.toString());
+                    }
+                }
+            }
+
+            @Override
+            public void failure(RetrofitError retrofitError) {
+                System.out.println(retrofitError.getBody());
+            }
+        };
+
+        LocationManager locationManager = (LocationManager) this.getSystemService(Context.LOCATION_SERVICE);
+
+        // Define a listener that responds to location updates
+        LocationListener locationListener = new LocationListener() {
+            public void onLocationChanged(Location location) {
+                System.out.println(location.getLatitude() + ", " + location.getLongitude());
```

Comment:
@intrications, you are right. How can I get the the location only once? We could get the position once the user start the app, and add a refresh button to manually reload places based on a new position.